aspnetapp/README.md

```diff
@@ -1,36 +1,109 @@
 
-Instructions
-------------
+This sample requires [Docker 17.05](https://docs.docker.com/release-notes/docker-ce/#17050-ce-2017-05-04) or later of the [Docker client](https://www.docker.com/products/docker). You need the latest Windows 10 or Windows Server 2016 to use [Windows containers](http://aka.ms/windowscontainers). The instructions assume you have the [Git](https://git-scm.com/downloads) client installed.
 
-First, prepare your environment by cloning the repository and navigating to the sample:
+## Getting the sample
+
+The easiest way to get the sample is by cloning the samples repository with git, using the following instructions. You can also just download the repository (it is small) as a zip from the [.NET Core Docker samples](https://github.com/dotnet/dotnet-docker-samples/) respository.
 
 ```console
 git clone https://github.com/dotnet/dotnet-docker-samples/
-cd dotnet-docker-samples/aspnetapp
 ```
 
-Follow these steps to run the sample locally:
+## Build and run the sample with Docker for Linux containers
+
+You can build and run the sample in Docker using Linux containers using the following commands. The instructions assume that you are in the root of the repository.
 
 ```console
-dotnet restore
-dotnet run
+cd aspnetapp
+docker build -t aspnetapp .
+docker run -it --rm -p 8000:80 --name aspnetcore_sample aspnetapp
 ```
 
-Follow these steps to run this sample in a Linux or Windows container:
+After the application starts, visit `http://localhost:8000` in your web browser.
+
+Note: The `-p` argument maps port 8000 on you local machine to port 80 in the container (the form of the port mapping is `host:container`). See the [Docker run reference](https://docs.docker.com/engine/reference/commandline/run/) for more information on commandline paramaters.
+
+## Build and run the sample with Docker for Windows containers
+
+You can build and run the sample in Docker using Windows containers using the following commands. The instructions assume that you are in the root of the repository.
 
 ```console
+cd aspnetapp
 docker build -t aspnetapp .
-docker run -d -p 80:80 aspnetapp
+docker run -it --rm --name aspnetcore_sample aspnetapp
+```
+
+You must navigate to the container IP (as opposed to http://localhost) in your browser directly when using Windows containers. You can get the IP address of your container with the following steps:
```

Comment:
So it only works in _some_ RS2 installations. Great. The confusing thing is these steps print out a blank line instead of the container ip address if you _do_ have the windows build with this fix. I suppose we can improve the instructions later when we have more details.